Mar. 2021 news (#17276)
idaholab/moose: modules/doc/content/newsletter/2021_03.md

If you are using CIVET for testing and/or MooseDocs for documentation, please reference the
paper listed here: [citing.md#testing-and-documentation].

## Early Construction of Subapps of MultiApps

A design change on MultiApp has been made recently. With this change, MOOSE constructs
the sub-applications (subapps) of MultiApps by default right after when MOOSE adds the
MultiApps to the main application. Previously, the construction of subapps happens
within the initial setup of the main application after the execution of all Actions.
Derived MultiApps can set the private parameter "use_positions" of MultiApp to *false*
to override this behavior and delay the construction as before, which is a feature
required by the sampler MultiApps in the MOOSE stochastic module. This change should
not affect the calculations with MultiApps but allows the main application to inquiry
the subapps for setting up its objects such as kernels, boundary conditions, etc.
accordingly. The design is documented in [MultiApps](syntax/MultiApps/index.md) along
with some other document updates related with MultiApp.
=======

> >>>>>>> 26afefa6e4 (Mar. 2021 news)

Reviewer: Looks like a leftover from rebasing...

Author: @fdkong Can you fix this when you add your items.

Reviewer 2: I will

  syntax pages from another project without having the syntax registered. The page will simply state
  that the object is not available. This negates the need to exclude specific files from the content,
  which was prone to errors and maintenance problems.
- Eigenvalue executioner did not work when no power iterations were used since the SLEPc EPS solver
  was not parsed correctly from the MOOSE side.  Only one outer iteration is allowed in EPS when
  Newton/PJFNK/JFNK is chosen.  We updated the eigenvalue executioner to set the maximum iteration
  of EPS to one when one of the Newton-based eigensolvers is selected. Now, the eigenvalue
  executioner works appropriately when no power iterations are involved.

## Conservative Transfer Enhancement

The conservative transfer can conserve a physics field by adjusting the transferred solution
with a user-specified postprocessor.  An execute-on option (`execute_on=transfer`) for the
postprocessor is required to transfer the solution correctly. But, it might not be uncommon
that users forget to specify such an input parameter.  Besides, a moose-based application might

> hardcode the execute-on options so that users can not set the transfer option. We will check if

Reviewer 3: ```suggestion
hardcode the execute-on options so that users can not set the transfer option. The code will check if
```

> users have a correct execute-on option in the input file to fix these issues. We will error out

Reviewer 3: ```suggestion
users have a correct execute-on option in the input file to fix these issues.
```

> if users do not have the input file option while users try to use a conservative transfer.

Reviewer 3: ```suggestion
If users do not have the correct `execute_on` option when using a conservative transfer, the code will error out.
```

## Use Parallel Push/Pull in MultiAppTransfer

In a multiapp field transfer such as a nearest-node transfer or a projection transfer,

> parallel communication is needed for obtaining interpolation values.  The parallel

Reviewer 3: ```suggestion
parallel communication is needed to obtain interpolation values.  The parallel
```

MOOSE. If you use automatic differentiation in your application, please
reference the paper listed here: [citing.md#automatic-differentiation].

## Early Construction of Subapps of MultiApps

A design change on MultiApp has been made recently. With this change, MOOSE constructs
the sub-applications (subapps) of MultiApps by default right after when MOOSE adds the
MultiApps to the main application. Previously, the construction of subapps happens
within the initial setup of the main application after the execution of all Actions.
Derived MultiApps can set the private parameter "use_positions" of MultiApp to *false*
to override this behavior and delay the construction as before, which is a feature
required by the sampler MultiApps in the MOOSE stochastic module. This change should
not affect the calculations with MultiApps but allows the main application to inquiry
the subapps for setting up its objects such as kernels, boundary conditions, etc.
accordingly. The design is documented in [MultiApps](syntax/MultiApps/index.md) along
with some other document updates related with MultiApp.

## Bug Fixes and Minor Changes

> - Fix Stochastic Tools to provide a correct Latin Hypercube sample, this also included the addition

Reviewer 3: ```suggestion
- Fix Stochastic Tools to provide a correct Latin Hypercube sampler, this also included the addition
```
